packages/remirror__extension-entity-reference/src/entity-reference-extension.ts

```diff
@@ -103,6 +105,37 @@ export class EntityReferenceExtension extends MarkExtension<EntityReferenceOptio
     };
   }
 
+  /**
+   * Track click events passed through to the editor.
+   */
+  createEventHandlers(): CreateEventHandlers {
+    return {
+      /**
+       * listens to click events and call the "onClickMark" handler with any of:
+       * 1. no argument if the text clicked is not an highlight mark
+       * 2. the id of the clicked highlight mark
+       * 3. id of the shortest highlight mark in case of overlapping highlights
+       */
+      clickMark: (_event, clickState) => {
+        const { markRanges } = clickState;
+
+        if (markRanges.length === 0) {
+          return this.options.onClickMark();
+        }
+
+        let shortestMark = markRanges[0];
+
+        markRanges.forEach((mark) => {
+          if (mark.to - mark.from < shortestMark.to - shortestMark.from) {
```

Comment:
Can we reuse here the [getShortestEntityReference](https://github.com/remirror/remirror/blob/main/packages/remirror__extension-entity-reference/src/utils/shortest-entity-reference.ts)?